src/Symfony/Component/HttpClient/Retry/RetryStrategyInterface.php

```diff
@@ -23,5 +27,10 @@ interface RetryDeciderInterface
      *
      * @return ?bool Returns null to signal that the body is required to take a decision
      */
-    public function shouldRetry(string $requestMethod, string $requestUrl, array $requestOptions, int $responseStatusCode, array $responseHeaders, ?string $responseContent): ?bool;
+    public function shouldRetry(AsyncContext $context, ?string $responseContent, ?TransportExceptionInterface $exception): ?bool;
```

Comment:
`TransportExceptionInterface $exception` is always null from `RetryableHttpClient` implementation

Reply:
Sure, this anticipates #38426